File: frontend/src/services/email-log.service.ts
import { api } from './api';
import type { BasePagination, BaseResponse } from '@/types/api.types';
import type { EmailLog, EmailLogStatus } from '@/types/email-log.types';

export const emailLogService = {
  list: async (params: { page?: number; limit?: number; status?: EmailLogStatus; templateCode?: string }) =>
    (await api.get<BaseResponse<BasePagination<EmailLog>>>('/email-logs', { params })).data.data,
  get: async (id: number) =>
    (await api.get<BaseResponse<EmailLog>>(`/email-logs/${id}`)).data.data,
  retry: async (id: number) =>
    (await api.post<BaseResponse<{ logId: number; status: EmailLogStatus; retriedAt: string }>>(`/email-logs/${id}/retry`)).data.data,
};

File: backend/src/main/java/EazyTech/EazyHire/services/EmailLogService.java
package EazyTech.EazyHire.services;

import EazyTech.EazyHire.core.PaginationRequest;
import EazyTech.EazyHire.models.dtos.EmailLogDTO;
import EazyTech.EazyHire.models.dtos.EmailLogRetryResponseDTO;
import EazyTech.EazyHire.models.enums.EmailLogStatus;
import org.springframework.data.domain.Page;

public interface EmailLogService {
    Page<EmailLogDTO> list(Long companyId, EmailLogStatus status, String templateCode, PaginationRequest paginationRequest);

    EmailLogDTO get(Long companyId, Long logId);

    EmailLogDTO record(Long companyId, Long applicationId, String recipientEmail, String templateCode,
                       String subject, String bodyHtml, boolean sent);

    EmailLogRetryResponseDTO retry(Long companyId, Long userId, Long logId);
}
